### PR TITLE
TIP-706: Enhance family sorter to sort on localized labels.

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

Enhances the family normalizer and filter so that the PQB can filter the products based on the localized family labels.

What this means is that for each product we index it's family code and the associated labels for each activated languages.
I'm open to the discussion because that means that each product of the same family will index the same family information (that maybe cumbersome ?)

- [X] The family filter now checks if the provided family code exists. if not, it throws an `\ObjectNotFoundException`.
- [X] Some families labels have been added to the technical and technical_sql integration fixtures.

**ES Filter Checklist:**
- [X] Review the PQB filter integration tests
- [X] Add index mapping integration tests 
- [X] Write the filter & Spec
- [X] Update the documentation in `search_specification.rst` (made in this PR https://github.com/jjanvier/pim-community-dev/pull/133)
- [X] Is the PIM installable ?

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Ok
| Added Behats                      | -
| Added integration tests           | Ok
| Changelog updated                 | -
| Review and 2 GTM                  | -
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Sorter/FamilySorterIntegration.php [hunk 1]
```diff
@@ -20,34 +20,41 @@ protected function setUp()
...
+
+        $result = $this->executeSorter([['family', Directions::DESCENDING, ['locale'=> 'fr_FR']]]);
+        $this->assertOrder($result, ['fooA2', 'fooA', 'fooA1', 'baz']);
+    }
+
+    public function testSortCodeAscendant()
     {
         $result = $this->executeSorter([['family', Directions::ASCENDING]]);
-        $this->assertOrder($result, ['foo1', 'foo2', 'fooA', 'fooB', 'baz']);
+        $this->assertOrder($result, ['fooA', 'fooA1', 'fooA2', 'baz']);
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Sorter/FamilySorterIntegration.php:48]: shouldn't we test with data that contain a different order for labels and codes?

File: src/Pim/Component/Catalog/tests/integration/Normalizer/Indexing/ProductIndexingIntegration.php [hunk 2]
```diff
@@ -84,16 +84,16 @@ public function testProductWithAllAttributes()
             'identifier'    => 'foo',
             'created'       => $date->format('c'),
             'updated'       => $date->format('c'),
-            'family'        => 'familyA',
+            'family'        => [
+                'code'   => 'familyA',
+                'labels' => [],
```
Comment (jjanvier) [inline src/Pim/Component/Catalog/tests/integration/Normalizer/Indexing/ProductIndexingIntegration.php]: shouldn't we add some labels in technical sql?

**Review by jjanvier — APPROVED** [review]

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Sorter/FamilySorterIntegration.php (hunk 1 above)
Comment (SamirBoulil) [reply src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Sorter/FamilySorterIntegration.php:48]: Those families surely have labels and code (in the fixtures I added them). Or I don't understand your question.

File: src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/Field/FamilyFilter.php [hunk 3]
```diff
@@ -53,31 +59,31 @@ public function addFieldFilter(
             case Operators::IN_LIST:
                 $clause = [
                     'terms' => [
-                        self::FAMILY_KEY => $value
-                    ]
+                        'family.code' => $value,
```
Comment (damien-carcel) [inline src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/Field/FamilyFilter.php:62]: :+1:

**Review by damien-carcel — APPROVED** [review]

**Pull request merged**